h/static/scripts/directive/test/annotation-test.js

```diff
@@ -404,28 +255,25 @@ describe('annotation', function() {
     var $timeout;
     var $window;
     var fakeAnnotationMapper;
-    var fakeAnnotationUI;
-    var fakeDocumentDomainFilter;
-    var fakeDocumentTitleFilter;
     var fakeDrafts;
-    var fakeFeatures;
     var fakeFlash;
     var fakeGroups;
-    var fakeMomentFilter;
     var fakePermissions;
-    var fakePersonaFilter;
     var fakeSession;
-    var fakeSettings;
-    var fakeTags;
     var fakeTime;
-    var fakeUrlEncodeFilter;
     var sandbox;
 
     function createDirective(annotation) {
-      annotation = annotation || defaultAnnotation();
+      annotation = annotation || fixtures.defaultAnnotation();
+
+      // A new annotation won't have any saved drafts yet.
+      if (!annotation.id) {
+        fakeDrafts.get.returns(null);
+      }
```

Comment:
Also, the logic was duplicated in a couple of different fixtures previously. Now it is centralized in one place.